examples/hooks/components/layout/Tabs.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { cx } from '../../cx';
+
+import './Tabs.css';
+
+export type TabProps = {
+  children: React.ReactNode;
+  title: string;
+};
+
+const getTabKey = (index: number, suffix?: string) =>
+  [`tab-${index}`, suffix].filter(Boolean).join('-');
+
+export function Tabs({ children }) {
+  const [currentTab, setCurrentTab] = useState(0);
+  const tabsRefs = useRef<HTMLElement[]>([]);
+
+  useEffect(
+    () => tabsRefs.current && tabsRefs.current[currentTab].focus(),
+    [currentTab]
+  );
+
+  const onKeyDown = (event: React.KeyboardEvent) => {
+    let tabIndex;
+    switch (event.key) {
+      case 'ArrowLeft':
+        tabIndex = Math.max(0, currentTab - 1);
+        break;
+      case 'ArrowRight':
+        tabIndex = Math.min(currentTab + 1, React.Children.count(children) - 1);
+        break;
+      default:
+        return;
+    }
+
+    setCurrentTab(tabIndex);
+  };
+
+  return (
+    <div className="Tabs">
+      <div role="tablist" className="Tabs-header">
+        {React.Children.map<React.ReactChild, React.ReactElement<TabProps>>(
+          children,
+          (child, index) => {
+            const isSelected = currentTab === index;
+            return (
+              <button
+                role="tab"
+                aria-selected={isSelected}
+                aria-controls={getTabKey(index, 'item')}
+                id={getTabKey(index, 'title')}
+                tabIndex={isSelected ? 0 : -1}
+                className={cx('Tabs-title', isSelected && 'Tabs-title--active')}
+                ref={(element) => (tabsRefs.current[index] = element!)}
```

Comment:
Not sure you need a callback ref. If `tabsRef` was already an array of the right amount of `null` values, you could do `ref={tabsRef[index]}`.

Reply:
I was not able to make it work with this syntax. I tried to simplify the use case but no luck: https://codesandbox.io/s/prod-glitter-bvenf?file=/src/Wrapper.js.

Reply:
My initial suggested code was wrong. Here's a working sandbox: https://codesandbox.io/s/laughing-snowflake-vm92d

Reply:
I tried it and apparently all the refs in the array are set to the last child.

![image](https://user-images.githubusercontent.com/154633/145289102-34425786-9bbb-41e2-84ef-c7108c4bcf4a.png)

I looked into it but could not figure out why 😕.

Reply:
No it's normal, my sandbox actually doesn't work since hooks can't be used in loops. I guess having a single ref array and assigning it with a callback ref is the only way 🙂